### ISSUE TITLE
Vim does not always split windows with the desired height N specified in commands like :[N]split

### ISSUE THREAD

**Issue by bfrg** [vim/vim#5037]
**Describe the bug**
Vim does not always open a new split window with the desired height that was specified before a split command like `:botright [N]split`. This depends on the height of the above (or below) window where the new split will be opened. For example, if a `quickfix` window with 10 lines height is open at the bottom, `:botright 10new` won't have the desired height 10. But running `:botright 8new` will open a new split with 8 lines.

**To Reproduce**
1. Run `vim --clean`
2. Open a new split at the bottom: `:botright 10new`
3. Open a third window with the same height: `:botright 10new`
4. Result: the third window won't have the specified height 10. All three windows will have the same height. See screenshot 1.
5. Close all splits and redo step 2, but fix the height of the second window: `:botright 10new | setlocal winfixheight`. 
6. Open a third split: `:botright 10new`. 
7. Result: The second window will keep its height 10, but the bottom window won't have the specified height 10. Instead it will have the same height as the top one. See screenshot 2.
8. This behavior does not occur when the third window's height is set to be 2 lines smaller than the window below which it will be placed. Thus, running `:botright 8new` will open a new window at the bottom with 8 lines. See screenshots 3 and 4.
9. A workaround is to run something like `:botright 8new | resize 10` to obtain the desired height.

**Expected behavior**
Vim should always open the new split window with the specified height N.

**Screenshots**

| `set nowinfixheight` | `set winfixheight` |
| ---------------- | ---------------- |
| **Screenshot 1** ![screenshot-1](https://user-images.githubusercontent.com/6266600/66524841-df8ca600-eaf3-11e9-993b-da868df36090.png) | **Screenshot 2** ![screenshot-2](https://user-images.githubusercontent.com/6266600/66526123-e0bfd200-eaf7-11e9-96af-f963f815c57a.png) |
| **Screenshot 3** ![screenshot-3](https://user-images.githubusercontent.com/6266600/66524977-4316d380-eaf4-11e9-97e7-c8b899306111.png) | **Screenshot 4** ![screenshot-4](https://user-images.githubusercontent.com/6266600/66526475-16b18600-eaf9-11e9-8d56-ab0b9a97d2d1.png) |

**Environment (please complete the following information):**
 - Vim version `8.1.2115`
 - Gentoo Linux
 - xfce4-terminal and GVim

**Comment by andymass** [comment]
Is equalalways set?

**Comment by bfrg** [comment]
Yes, it is set by default in `vim --clean`. But why should this matter? In screenshot 3 and 4 you can see that the split has the correct height.

Looking at screenshot 3, it seems like Vim is drawing the new window "on top of" `window 2`. My guess is that's why opening a new split with 8 lines works fine since windows must have at least one line plus another line for the `statusline`. If `window 2` initially had 12 lines, `:botright 10new` would have worked fine, but `:botright 11new` would - similar to screenshot 1 and 2 - readjust the height of all three windows (unless `winfixheight` is set). 

If you run instead `:botright 1new | resize 12`, `window 2` will be pushed up to make enough room for the new split. I would have expected the same behavior when running `:botright 12new`.

**Comment by andymass** [comment]
My interpretation is steps 1--7 and screenshots 1 and 2 are WAE because equalalways equalizes the windows whenever a split happens (except for the winfixheight). That being said, this maybe should be changed because specifying a height for split is pretty useless then.

For screenshots 3 and 4, apparently equalalways isn't always applied when there is enough room in the above window a-priori.  It's only used when a window would become too small?

Simple fix is to temporarily mark the new window to be ignored by equalalways when a height is specified.  But regardless these rules need to be clarified in the help.

**Comment by bfrg** [comment]
But even with `set noequalaways` Vim equalizes all windows. 
1. Run `vim --clean -c "set noequalalways"`
2. `:botright 10new`
3. `:botright 10new`

Result:

| `:botright 10new` | `:botright 10new \| botright 10new` |
| ----------------------- | ------------------------------------------ |
| ![screenshot-2019-10-10_152449](https://user-images.githubusercontent.com/6266600/66572912-1dc8aa80-eb72-11e9-9ea1-6fab04d14fa5.png) | ![screenshot-2019-10-10_152342](https://user-images.githubusercontent.com/6266600/66572929-23be8b80-eb72-11e9-8401-feffb44c0348.png) |

**Comment by andymass** [comment]
Well, noequalalways doesn't exactly mean equalnever.  I read the code and vim does equalize the windows in some situations.  What vim could do instead is take height from the first window when the second window isn't sufficiently tall.

Alternatively, vim could equalize everything else but keep the new window the specified size.

**Comment by bfrg** [comment]
> What vim could do instead is take height from the first window when the second window isn't sufficiently tall.

This is exactly the behavior of `:botright 1new | resize 10`. I would have expected the same result when running `:botright 10new` directly.

**Comment by bfrg** [comment]
Related to #2999.
